### PR TITLE
add CLI play-playlist option, to play given playlist name.

### PR DESCRIPTION
**Pull request by NicolasToussaint**
Adding a command line option to start playing the playlist given as argument.

I reused the command line argument playlist_name_ from the create_playlist option - I had to add it to the [De]Serialize functions to make it work, meaning -unless I missed something- that the create playlist could not be working before.
And this "create playlist" is still not working for me, but I did not dig further.

### CODE REVIEW

File: src/core/player.cpp [hunk 1]
```diff
@@ -224,6 +224,34 @@ void Player::NextItem(Engine::TrackChangeFlags change) {
   PlayAt(i, change, false);
 }
 
+void Player::PlayPlaylist(const QString& playlistName) {
+  PlayPlaylistInternal(Engine::Manual, playlistName);
+}
+
+void Player::PlayPlaylistInternal(Engine::TrackChangeFlags change,
+                                  const QString& playlistName) {
+  Playlist* playlist = NULL;
```
Comment (hatstand) [inline src/core/player.cpp:233]: Use `nullptr` instead of `NULL`.

File: src/core/player.cpp [hunk 2]
```diff
@@ -224,6 +224,34 @@ void Player::NextItem(Engine::TrackChangeFlags change) {
...
+  for (Playlist* p : app_->playlist_manager()->GetAllPlaylists()) {
+    if (playlistName == app_->playlist_manager()->GetPlaylistName(p->id())) {
+      playlist = p;
+      break;
+    }
+  }
```
Comment (hatstand) [inline src/core/player.cpp:239]: You can probably write this `for` loop as a one liner with `std::find`

File: src/core/player.cpp [hunk 3]
```diff
@@ -224,6 +224,34 @@ void Player::NextItem(Engine::TrackChangeFlags change) {
...
+
+  if (playlist != NULL) {
```
Comment (hatstand) [inline src/core/player.cpp:241]: Invert this logic so it returns early if no playlist was found.

File: src/ui/mainwindow.cpp [hunk 4]
```diff
@@ -2270,6 +2271,14 @@ void MainWindow::CommandlineOptionsReceived(const CommandlineOptions& options) {
     case CommandlineOptions::Player_Next:
       app_->player()->Next();
       break;
+    case CommandlineOptions::Player_PlayPlaylist:
+      if (options.playlist_name().isNull() ||
+          options.playlist_name().isEmpty()) {
```
Comment (hatstand) [inline src/ui/mainwindow.cpp:2276]: `isEmpty()` is sufficient.

File: src/core/player.cpp (hunk 1 above)
Comment (NicolasToussaint) [reply src/core/player.cpp:233]: OK

File: src/core/player.cpp (hunk 2 above)
Comment (NicolasToussaint) [reply src/core/player.cpp:239]: sorry, that's reaching out of my C++ skills :-/

**Pull request merged**